components/tikv_util/src/security.rs

```diff
@@ -20,6 +21,7 @@ pub struct SecurityConfig {
     #[serde(skip)]
     pub override_ssl_target: String,
     pub cipher_file: String,
+    pub x509_common_names: HashSet<String>,
```

Comment:
I suggest to use String instead and require all certificates to share the same common name. Because we don't validate roles at all, so connecting a PD who claims itself as TiDB is totally allowed in current implementations, which can be very confusing. Note that all other similar products also use a string instead of set.

Reply:
Switch back to HashSet